doc/configuration.rst

```diff
@@ -957,41 +957,42 @@ you can do::
 Controlling what output is captured
 ===================================
 
-Sphinx-Gallery can run your code blocks in two ways:
+The ``print_eval_repr`` configuration allows the user to control what output of
+the last expression in a code block, is captured and rendered in the built
+documentation. By default ``print_eval_repr`` is an empty tuple. With this
+setting, the last statement in a code block is executed with ``exec()`` and
+thus only data directed to standard output is captured.
 
-* **Exec only** - ``exec()`` the whole code block. Only data directed to
-  standard output is captured.
-* **Exec eval** - ``exec()`` everything in the code block *except* the last
-  statement. If the last statement is an expression, ``eval()`` it and if it is
-  not an expression, ``exec()`` it. Both data directed to standard output
-  and the value of the last statement, if it is an expression, is captured.
+If you added ``__repr__`` to the tuple like so::
+
+        sphinx_gallery_conf = {
+            ...
+            'print_eval_repr': ('__repr__',),
```

Comment:
I would make the example `('__html_repr__', '__repr__')` and describe that it will go through the tuple in order and try to execute one that returns a `str` (which `__repr__` should always do) so in this case something will always be renderered for a non-assigment last line.